salesagility/SuiteCRM
# Hotfix statechecker updates

## Description




Adding state saver and checker changes in 7.8.x. see more in related PR: https://github.com/salesagility/SuiteCRM/pull/5881

- update state saver and state checker

- verbose phpunit tests
(TODO: how to run phpunit tests only)

- codecept settings for netbeans
(TODO: hot to setup unit tests and coverage report)

- code and tests clean up

## Motivation and Context


- state saver and checker is updated
- developers want to create and run tests and code coverage report

## How To Test This


- Install SuiteCRM
- Run unit test

Extra benefit for NetBeans users:

Project setup:
![image](https://user-images.githubusercontent.com/13831178/40067757-d4618b0c-585e-11e8-98ca-c384af311beb.png)
![image](https://user-images.githubusercontent.com/13831178/40067744-c9100152-585e-11e8-8b21-aa0ec3236a94.png)

Run tests:
Right click on project name -> Test (Alt-F6)
Additional parameters for codeception:
![image](https://user-images.githubusercontent.com/13831178/40067899-35478f20-585f-11e8-9253-23c8df057453.png)
see more about codeception parameters: 
https://docs.suitecrm.com/developer/appendix-c---automated-testing/#_writing_and_running_tests

Collect Coverage:
Right click on project name -> Code Coverage -> Collect and Display Code Coverage
Run tests
See the covered and uncovered (green/red) code in your IDE:
![image](https://user-images.githubusercontent.com/13831178/40067791-e991665a-585e-11e8-8d2a-d3dd806d18e3.png)

See full Code Coverage Report for SuiteCRM in your IDE:
![image](https://user-images.githubusercontent.com/13831178/40067827-fd3904d8-585e-11e8-9882-32f58110895f.png)

(TODO: this part needs to update documentation for SuiteCRM 7.10.x only)

## Types of changes


### Final checklist
